lib/arqo/query.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module ARQO
+  # Parent class for query objects
+  class Query
+    attr_reader :relation
+    delegate_missing_to :relation
+
+    def initialize(relation = derived_relation)
+      @relation = relation
+      @relation = @relation.extending(scope_module) if Object.const_defined?(scope_module_name)
+    end
+
+    private
+
+    def derived_relation_name
+      self.class.name.sub(/Query$/, '')
+    end
```

Comment:
It'd be nice to allow the end user to configure the class name associated to the query object in case they want to use a different name than the model.

Reply:
nice, I'll create a ticket for it